File: java-kotlin/AOP-Java/src/main/java/org/example/aopjava/controller/ExampleController.java
package org.example.aopjava.controller;

import lombok.extern.slf4j.*;
import org.example.aopjava.service.*;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/aop/")
@Slf4j
public class ExampleController {

    private final ExampleService exampleService;

    public ExampleController(ExampleService exampleService) {
        this.exampleService = exampleService;
    }

    @GetMapping("example")
    public String exampleMethodController() {
        return exampleService.exampleMethod();
    }

    @GetMapping("async")
    public String exampleMethodAsyncController() {
        exampleService.exampleMethodAsync();
        log.info("exampleMethodAsyncController called");
        return "Async method called";
    }
}

File: java-kotlin/kotlin-rest-springboot/src/main/kotlin/org/example/kotlinrestspringboot/controller/ExampleController.kt
package org.example.kotlinrestspringboot.controller


import org.example.kotlinrestspringboot.model.ModelRequest
import org.example.kotlinrestspringboot.model.ModelResponse
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.*
import org.springframework.web.multipart.MultipartFile


@RestController
@RequestMapping("api/v1/example")
class ExampleController {

    private val log = LoggerFactory.getLogger(ExampleController::class.java)

    @GetMapping("/get/{id}")
    fun get(@PathVariable id: String): String {
        log.info("Get method with id: $id")
        return "GET method"
    }

    @PostMapping("/post")
    fun post(@RequestBody req: ModelRequest?): ModelResponse {
        log.info("Post method with req: ${req?.req}")
        return ModelResponse("200")
    }

    @PatchMapping("/patch")
    fun patch(@RequestParam data: ModelRequest?, @RequestParam files: Array<MultipartFile>?): ModelResponse {
        log.info("Patch method with req: ${data?.req} total file: ${files?.size}")
        return ModelResponse("200")
    }

    @GetMapping("/get/error")
    fun getError(): String {
        log.info("Get method with error")
        throw RuntimeException("Error")
    }
}